src/Word.java

```diff
@@ -1,78 +1,63 @@
 import java.util.HashSet;
 import java.util.Set;
+import java.util.stream.Collectors;
 
 public class Word {
-    private Set<Character> guessedChars;
-    private String originalWord;
 
-    public Word(String originalWord) {
-        this.originalWord = originalWord;
-        guessedChars = new HashSet<>();
-    }
-
-    private static Set<Character> unguessableChars = new HashSet<>();
-    {
-        unguessableChars.add(' ');
-        unguessableChars.add('"');
-        unguessableChars.add('.');
-        unguessableChars.add(',');
-        unguessableChars.add('@');
-    }
+  private Set<Character> guessedChars;
+  private String originalWord;
 
-    public GuessResult guess(String guess) {
-        if (guess.length() == 1) {
-            char c = guess.charAt(0);
+  Word(String originalWord) {
+    this.originalWord = originalWord;
+    guessedChars = new HashSet<>();
+  }
 
-            if (unguessableChars.contains(c)) {
-                return GuessResult.UNGUESSABLE_CHAR;
-            }
+  public int wordLength(){
+    return (int) originalWord.chars().mapToObj(i -> (char) i).filter(Character::isLetter).count();
+  }
 
-            if (guessedChars.contains(c)) {
-                return GuessResult.CHAR_ALREADY_GUESSED;
-            }
 
-            guessedChars.add(c);
-
-            if (originalWord.contains(c + "")) {
-                Set<Character> allChars = new HashSet<>(guessedChars);
-                allChars.addAll(unguessableChars);
+  public GuessResult guess(String guess) {
+    if (guess.length() == 0) {
+      return GuessResult.UNGUESSABLE_CHAR;
+    }
+    if (guess.length() == 1) {
+      char c = Character.toLowerCase(guess.charAt(0));
 
-                for (int i = 0; i < originalWord.length(); i++) {
-                    if (!allChars.contains(originalWord.charAt(i))) {
-                        return GuessResult.CORRECT_GUESS;
-                    }
-                }
+      if (!Character.isLetter(c)) {
+        return GuessResult.UNGUESSABLE_CHAR;
+      }
 
-                return GuessResult.COMPLETED_WORD;
-            }
+      if (guessedChars.contains(c)) {
+        return GuessResult.CHAR_ALREADY_GUESSED;
+      }
 
-            return GuessResult.INCORRECT_GUESS;
-        }
+      guessedChars.add(c);
 
-        if (originalWord.equals(guess)) {
-            return GuessResult.COMPLETED_WORD;
+      if (originalWord.contains(c + "") || originalWord.contains(Character.toUpperCase(c) + "")) {
+        if (guessedChars.containsAll(originalWord.chars().mapToObj(i -> Character.toLowerCase((char) i))
```

Comment:
I'm really starting to dislike all this stream magic. It's a lot less readable, so why change it? :)